snap/snapenv/snapenv.go

```diff
@@ -120,13 +129,15 @@ func basicEnv(info *snap.Info) map[string]string {
 func userEnv(info *snap.Info, home string) map[string]string {
 	// TODO parallel-install: use of proper instance/store name
 	result := map[string]string{
-		"SNAP_USER_COMMON": info.UserCommonDataDir(home),
-		"SNAP_USER_DATA":   info.UserDataDir(home),
-		"XDG_RUNTIME_DIR":  info.UserXdgRuntimeDir(sys.Geteuid()),
+		"SNAP_USER_COMMON":          snap.UserCommonDataDir(home, info.SnapName()),
+		"SNAP_USER_DATA":            snap.UserDataDir(home, info.SnapName(), info.Revision),
+		"SNAP_INSTANCE_USER_COMMON": info.UserCommonDataDir(home),
+		"SNAP_INSTANCE_USER_DATA":   info.UserDataDir(home),
+		"XDG_RUNTIME_DIR":           info.UserXdgRuntimeDir(sys.Geteuid()),
 	}
 	// For non-classic snaps, we set HOME but on classic allow snaps to see real HOME
 	if !info.NeedsClassic() {
-		result["HOME"] = info.UserDataDir(home)
+		result["HOME"] = snap.UserDataDir(home, info.SnapName(), info.Revision)
```

Comment:
Since we aren't doing user mounts, this should be `info.SnapInstanceName()`, no?

Reply:
Actually, this was fixed later in https://github.com/snapcore/snapd/pull/5735. Thanks @bboozzoo for pointing this out.